### ISSUE TITLE
Add proper CMake file to skill trees skeleton code then make pull request

### ISSUE THREAD

**Issue by jackandthebean** [uchicago-cs/chiventure#561]
# Add proper CMake file to skill trees skeleton code then make pull request
### What
Our skeleton code consists of two files: `skilltrees.h` and `skilltrees.c`, which implement the structs that will hold chiventure-supported skills, as well as their associated new, init, and free functions. The skeleton code also implements basic functions that manipulate skill lists—adding/removing skills—and retrieve skill types that already exist within a given skill list. These files need a `CMakeLists.txt` to compile and progress into the PR phase.

### Why
Our skeleton code is ready for our first PR, #601.

**Comment by jackandthebean** [comment]
3a175ce

**Comment by jackandthebean** [comment]
I'm closing this issue, as the CMake files were added and our PR has been reviewed. I've now made the suggested changes to the PR (77e54f6) and look forward to making more extensive tests!

**Comment by jackandthebean** [comment]
I am closing this issue, as our skill trees skeleton code was successfully merged into `dev`! Besides making @firatciftci 's suggested changes, I reworked all the code to match the style of all our assignments more closely, increased the flexibility of the skill "effect" function pointers, and added a `skill_execute()` function which simply executes the function pointed to by a given `skill_t` struct. The flexibility of the skill effect functions was one of our team's most contentious issues, the solution for which I documented in #559. With this skeleton code merged into `dev`, hopefully the process of merging our more advanced functions will be more seamless; we look forward to making those future PRs!
